lesson17/index.js

```diff
@@ -0,0 +1,171 @@
+const url = "https://myjson.dit.upm.es/api/bins/8b9d";
+const main = document.querySelector("main");
+
+const controlSlide = (btnType) => {
+  const currentSlide = document.querySelector('[data-hidden="false"]');
+  if(btnType === "prev") {
+    // 今のcurrentのスライドの一つ前の要素をcurrentにする
+    const prevSlide = currentSlide.previousElementSibling;
+    currentSlide.setAttribute("data-hidden","true");
+    prevSlide.setAttribute("data-hidden","false");
+    controlBtnBehavior();
+  } else {
+    // 今のcurrentのスライドの一つ次の要素をcurrentにする
+    const nextSlide = currentSlide.nextElementSibling;
+    currentSlide.setAttribute("data-hidden","true");
+    nextSlide.setAttribute("data-hidden","false");
+    controlBtnBehavior();
+  }
+  updateFractionNum();
+}
+
+const slideImgTemplate = (index, slide) => {
+  const div = document.createElement("div");
+  div.classList.add("sliderImg");
+  div.id = `slide-${index}`;
+  if(index === 0){
+    div.setAttribute("data-hidden", "false");
+  } else {
+    div.setAttribute("data-hidden", "true");
+  }
+  const img = document.createElement("img");
+  img.src = slide.img;
+  div.appendChild(img);
+  return div;
+};
+
+const initSlides = (slides)=> {
+  const slider = document.createElement("div");
+  slider.classList.add("slider");
+  slider.id = "js-slider"
+  const sliderWrapper = document.createElement("div");
+  sliderWrapper.classList.add("sliderWrapper");
+  sliderWrapper.id = "js-sliderWrapper"
+  slider.appendChild(sliderWrapper);
+  main.appendChild(slider);
+
+  const fragment = document.createDocumentFragment();
+  for (const [index, slide] of slides.entries()) {
+    fragment.appendChild(slideImgTemplate(index, slide));
+    sliderWrapper.appendChild(fragment);
+  }
+  createSliderBtns();
+}
+
+const setBtnEvent = () => {
+  const prevBtn = document.getElementById("js-slider-prevBtn");
+  const nextBtn = document.getElementById("js-slider-nextBtn");
+  prevBtn.addEventListener("click", ()=> {
+    controlSlide('prev');
+  })
+  nextBtn.addEventListener("click", ()=> {
+    controlSlide('next');
+  })
+}
```

Comment:
IMO
I changed this function a bit along with the change in createSliderBtns. I hope this gives you a feel for it!
```suggestion
const setBtnEvent = () => {
  const buttons = document.querySelectorAll(".sliderBtn");
  buttons.forEach((button) => {
    button.addEventListener("click", (e) => {
      const eventTarget = e.target;
      controlSlide(eventTarget.value);
    });
  });
};
```

Reply:
[0264e28](https://github.com/hippopo-moo/js-handson-work/pull/19/commits/0264e289a690c5079efab546b3de3fcea2972439)

Thanks for this one too!
I've modified the code based on the code you provided.
Am I getting the intent right?